### PR TITLE
Add antigen@2.2.3

### PR DESCRIPTION
**Pull request by justforlxz**
Hi, this is the first time I have made an rpm package. Maybe something is wrong, I will try my best to modify it.

### CODE REVIEW

File: rpms/antigen/antigen.spec [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+%prep
+%setup -q
+
+%install
+make build
+make PREFIX=%{buildroot}/usr/share/zsh install
+
+%files
+%license LICENSE
+/usr
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec]: -  use [macros](https://docs.fedoraproject.org/en-US/packaging-guidelines/RPMMacros/#_macros_for_paths_set_and_used_by_build_systems)
 - Don't glob here, consider listing all files installed by this package.

File: rpms/antigen/antigen.spec [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Requires: git
+
+%description
+Antigen is a small set of functions that help you easily manage your shell (zsh) plugins, called bundles. The concept is pretty much the same as bundles in a typical vim+pathogen setup. Antigen is to zsh, what Vundle is to vim.
+
+%prep
+%setup -q
+
+%install
+make build
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec:23]: - build in `%build` section.
 - build with `%make_build`
 
But the makefile don't seem to do much work, `make build` could be fine

File: rpms/antigen/antigen.spec [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+make PREFIX=%{buildroot}/usr/share/zsh install
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec]: - you are installing to `/usr/share/zsh/share/antigen.zsh`, the path is weird
 - use [macros](https://docs.fedoraproject.org/en-US/packaging-guidelines/RPMMacros/#_macros_for_paths_set_and_used_by_build_systems)
 - considering using `%make_install`, which sets `PREFIX` for you.

File: rpms/antigen/antigen.spec [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+License: MIT
+URL: https://github.com/zsh-users/antigen
+Source0: https://github.com/zsh-users/antigen/releases/download/v%{version}/v%{version}.tar.gz
+BuildRequires: make
+Requires: zsh
+Requires: git
+
+%description
+Antigen is a small set of functions that help you easily manage your shell (zsh) plugins, called bundles. The concept is pretty much the same as bundles in a typical vim+pathogen setup. Antigen is to zsh, what Vundle is to vim.
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec]: This line is toooo long, split it.

File: rpms/antigen/antigen.spec [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+%prep
+%setup -q
+
+%install
+make build
+make PREFIX=%{buildroot}/usr/share/zsh install
+
+%files
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec:28]: consider adding `%doc  README.mkd ` to provide users a guide to use this package.

**Review by karuboniru — CHANGES_REQUESTED** [review]

File: rpms/antigen/antigen.spec (hunk 3 above)
Comment (justforlxz) [reply rpms/antigen/antigen.spec]: if i used %make_install, it seems that PREFIX is not set.

```
+ /usr/bin/make install DESTDIR=/home/lxz/rpmbuild/BUILDROOT/antigen-2.2.3-1.fc34.x86_64 'INSTALL=/usr/bin/install -p'
mkdir -p /usr/local/share && cp /home/lxz/rpmbuild/BUILD/antigen-2.2.3/bin/antigen.zsh /usr/local/share/antigen.zsh
cp: cannot create regular file '/usr/local/share/antigen.zsh': Permission denied
```

I want the file to be installed in /usr/share/antigen.zsh, I need to set PREFIX to %{_prefix}, which is /usr, how should I write it?  thx :)

Comment (karuboniru) [reply rpms/antigen/antigen.spec]: my mistake, just use `make PREFIX=%{buildroot}%{_prefix} install` is enough. Upstream makefile/configure can't handle normal installation progress.

**Comment by justforlxz** [comment]
@karuboniru  it's finished, please review again, thx!

File: rpms/antigen/antigen.spec [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+%prep
+%setup -q
+
+%install
+make build
+make PREFIX=%{buildroot}/%{_prefix} install
+
+%files
+%license LICENSE
+%{_prefix}/*
```
Comment (karuboniru) [inline rpms/antigen/antigen.spec]: %{_datadir}/antigen.zsh

If you use %{_prefix}/*,your package will own /usr/share.

**Review by karuboniru — CHANGES_REQUESTED** [review]

File: rpms/antigen/antigen.spec (hunk 2 above)
Comment (karuboniru) [reply rpms/antigen/antigen.spec:23]: `make build` is fine. but please separate building process into `%build` part
```
%build
make build

%install
make PREFIX=%{buildroot}/%{_prefix} install
```
@justforlxz

Comment (justforlxz) [reply rpms/antigen/antigen.spec:23]: if I have %build, the rpmbuild need debugsourcefiles.list, do I ignore the problem and use --nodebuginfo？

Comment (karuboniru) [reply rpms/antigen/antigen.spec:23]: `%global debug_package %{nil}`
add to the start of the spec file.

Comment (justforlxz) [reply rpms/antigen/antigen.spec:23]: done! thx!

**Pull request merged**